dali/python/nvidia/dali/plugin/base_iterator.py

```diff
@@ -202,6 +202,21 @@ def __init__(self,
         self._ever_scheduled = False
         self._ever_consumed = False
 
+        self._checkpointing = self._pipes[0]._checkpointing
+        for p in self._pipes:
+            assert p._checkpointing == self._checkpointing, \
+                   "All pipelines should have the same value of the checkpointing option"
+
+        if self._checkpointing:
+            assert self._last_batch_padded, \
+                   "Currently, checkpointing is not supported with last_batch_padded=False"
+            assert self._last_batch_policy != LastBatchPolicy.DROP, \
+                   "Currently, checkpointing is not supported with last_batch_policy=DROP"
+
+            # Precompute the initial checkpoints, to prevent any problems
+            # related to the `prepare_first_batch` flag.
+            self._initial_checkpoints = [p.checkpoint() for p in self._pipes]
```

Comment:
For user-facing validation, please use exceptions, for example `ValueErorr, NotImplementedError` instead of assertions.